changelog.d/15599.bugfix

```diff
@@ -0,0 +1 @@
+Print full error and stack-trace of any exception that occurs during startup/initialization.
```

Comment:
Using the same changelog as https://github.com/matrix-org/synapse/pull/15569/ so they merge